llama-index-integrations/tools/llama-index-tools-agentql/llama_index/tools/agentql/base.py

```diff
@@ -0,0 +1,230 @@
+from typing import Optional
+import os
+
+from playwright.async_api import Browser as AsyncBrowser
+
+from llama_index.core.tools.tool_spec.base import BaseToolSpec
+
+from llama_index.tools.agentql.const import (
+    DEFAULT_EXTRACT_ELEMENTS_TIMEOUT_SECONDS,
+    DEFAULT_EXTRACT_DATA_TIMEOUT_SECONDS,
+    DEFAULT_IS_STEALTH_MODE_ENABLED,
+    DEFAULT_WAIT_FOR_NETWORK_IDLE,
+    DEFAULT_INCLUDE_HIDDEN_DATA,
+    DEFAULT_INCLUDE_HIDDEN_ELEMENTS,
+    DEFAULT_RESPONSE_MODE,
+    DEFAULT_WAIT_FOR_PAGE_LOAD_SECONDS,
+    DEFAULT_IS_SCROLL_TO_BOTTOM_ENABLED,
+    DEFAULT_IS_SCREENSHOT_ENABLED,
+    DEFAULT_API_TIMEOUT_SECONDS,
+    REQUEST_ORIGIN,
+)
+from llama_index.tools.agentql.messages import (
+    QUERY_PROMPT_REQUIRED_ERROR_MESSAGE,
+    QUERY_PROMPT_EXCLUSIVE_ERROR_MESSAGE,
+    UNSET_API_KEY_ERROR_MESSAGE,
+    MISSING_BROWSER_ERROR_MESSAGE,
+)
+from llama_index.tools.agentql.load_data import aload_data
+from llama_index.tools.agentql.utils import (
+    _aget_current_agentql_page,
+    validate_url_scheme,
+)
+
+
+class AgentQLToolSpec(BaseToolSpec):
+    """
+    AgentQL tool spec.
+    """
+
+    spec_functions = [
+        "extract_web_data_from_rest_api",
```

Comment:
it was "with" before, and it is "with" in LC
```suggestion
        "extract_web_data_with_rest_api",
```